src/fluree/db/json_ld/transact.cljc

```diff
@@ -580,7 +580,10 @@
   [db fuel-tracker parsed-txn tx-state]
   (go
     (let [error-ch  (async/chan)
-          update-ch (->> (where/search db parsed-txn fuel-tracker error-ch)
+          ;; ensure at least one solution is on the solution-ch
+          solution-ch (async/merge [(async/to-chan [where/blank-solution])
```

Comment:
I think you should use the already defined [`with-default` transducer](https://github.com/fluree/db/blob/main/src/fluree/db/query/exec/where.cljc#L622) for this. If nothing passes through the channel, it will send the default solution through. This not only will result in less async channels, but it will also ensure that the blank solution is only passed through when there truly no other solutions instead of always. Here is [an example of it's use in the implementation of `:optional` patterns](https://github.com/fluree/db/blob/main/src/fluree/db/query/exec/where.cljc#L655). 

If there's no way for you to add it directly to a preexisting channel in the `modify2` function, then you can make a new channel with the transducer on it and pipe the channel returned from the `where/search` function into it.